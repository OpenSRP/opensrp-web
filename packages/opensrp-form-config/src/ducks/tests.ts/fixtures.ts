export const fixManifestReleases = [
    {
        identifier: '1.0.11',
        json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:00:38.413+02:00',
        updatedAt: '2020-06-08T15:00:38.413+02:00',
    },
    {
        identifier: '1.0.12',
        json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:03:00.950+02:00',
        updatedAt: '2020-06-08T15:03:00.950+02:00',
    },
    {
        identifier: '1.0.16',
        json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.16',
        createdAt: '2020-06-09T14:40:04.189+02:00',
        updatedAt: '2020-06-09T14:40:04.189+02:00',
    },
];
