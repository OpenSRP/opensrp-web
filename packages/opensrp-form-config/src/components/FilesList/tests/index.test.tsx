import React from 'react';
import { mount, shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import ConnectedManifestFilesList, { ManifestFilesList } from '../index';
import { getFetchOptions, OpenSRPService } from '@opensrp/server-service';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import reducerRegistry, { store } from '@onaio/redux-reducer-registry';
import flushPromises from 'flush-promises';
import reducer, { fetchManifestFiles, filesReducerName } from '../../../ducks/manifestFiles';
import { fixManifestFiles } from '../../../ducks/tests.ts/fixtures';

/** register the reducers */
reducerRegistry.register(filesReducerName, reducer);

const history = createBrowserHistory();

const baseURL = 'https://test-example.com/rest';
const props = {
    baseURL,
    downloadEndPoint: 'form-download',
    endpoint: 'form/related',
    fileUploadUrl: 'manifest',
    formVersion: '123',
    getPayload: getFetchOptions,
    isJsonValidator: true,
    LoadingComponent: <div></div>,
    uploadTypeUrl: 'manifest-file',
};

describe('components/manifestFiles', () => {
    it('renders without crashing', () => {
        shallow(<ManifestFilesList {...props} />);
    });

    it('renders without crashing when connected to store', async () => {
        store.dispatch(fetchManifestFiles(fixManifestFiles));
        const mockList = jest.fn();
        OpenSRPService.prototype.list = mockList;
        mockList.mockReturnValueOnce(Promise.resolve(fixManifestFiles));

        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <ConnectedManifestFilesList {...props} />
                </Router>
            </Provider>,
        );

        await flushPromises();
        wrapper.update();
        expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot();
        expect(wrapper.find('SearchBar').length).toEqual(1);
        expect(
            wrapper
                .find('Row Col')
                .at(1)
                .text(),
        ).toEqual('Upload New File');

        // to figure out why table is not being loaded
        // expect(wrapper.find('.tbody .tr').length).toEqual(fixManifestFiles.length);
    });
});
