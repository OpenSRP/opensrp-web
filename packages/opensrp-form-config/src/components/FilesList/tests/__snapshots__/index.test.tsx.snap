// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/manifestFiles renders without crashing when connected to store 1`] = `
Object {
  "CellComponent": [Function],
  "columns": Array [
    Object {
      "Header": "Identifier",
      "accessor": "identifier",
    },
    Object {
      "Header": "File Name",
      "accessor": "label",
    },
    Object {
      "Header": "File Version",
      "accessor": "version",
    },
    Object {
      "Header": "Edit",
      "accessor": [Function],
      "disableSortBy": true,
    },
    Object {
      "Header": " ",
      "accessor": [Function],
      "disableSortBy": true,
    },
  ],
  "data": Array [
    Object {
      "createdAt": "Jun 19, 2020, 4:23:22 PM",
      "form_relation": "",
      "id": "53",
      "identifier": "reveal-test-file.json",
      "isDraft": false,
      "isJsonValidator": false,
      "jursdiction": "",
      "label": "test publish",
      "module": "",
      "version": "1.0.27",
    },
    Object {
      "createdAt": "Jun 19, 2020, 12:31:37 PM",
      "form_relation": "",
      "id": "52",
      "identifier": "test-form-1.json",
      "isDraft": false,
      "isJsonValidator": false,
      "jursdiction": "",
      "label": "test form",
      "module": "",
      "version": "1.0.26",
    },
  ],
  "hasChildren": [Function],
  "identifierField": "id",
  "linkerField": "id",
  "loading": false,
  "loadingComponent": [Function],
  "paginate": false,
  "parentIdentifierField": "parent_id",
  "renderNullDataComponent": [Function],
  "resize": true,
  "rootParentId": null,
  "useDrillDown": false,
}
`;

exports[`components/manifestFiles renders without crashing when connected to store 2`] = `
<DrillDownTable
  CellComponent={[Function]}
  columns={
    Array [
      Object {
        "Header": "Identifier",
        "accessor": "identifier",
      },
      Object {
        "Header": "File Name",
        "accessor": "label",
      },
      Object {
        "Header": "File Version",
        "accessor": "version",
      },
      Object {
        "Header": "Edit",
        "accessor": [Function],
        "disableSortBy": true,
      },
      Object {
        "Header": " ",
        "accessor": [Function],
        "disableSortBy": true,
      },
    ]
  }
  data={
    Array [
      Object {
        "createdAt": "Jun 19, 2020, 4:23:22 PM",
        "form_relation": "",
        "id": "53",
        "identifier": "reveal-test-file.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test publish",
        "module": "",
        "version": "1.0.27",
      },
      Object {
        "createdAt": "Jun 19, 2020, 12:31:37 PM",
        "form_relation": "",
        "id": "52",
        "identifier": "test-form-1.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test form",
        "module": "",
        "version": "1.0.26",
      },
    ]
  }
  hasChildren={[Function]}
  identifierField="id"
  linkerField="id"
  loading={false}
  loadingComponent={[Function]}
  paginate={false}
  parentIdentifierField="parent_id"
  renderNullDataComponent={[Function]}
  resize={true}
  rootParentId={null}
  useDrillDown={false}
>
  <Table
    CellComponent={[Function]}
    columns={
      Array [
        Object {
          "Header": "Identifier",
          "accessor": "identifier",
        },
        Object {
          "Header": "File Name",
          "accessor": "label",
        },
        Object {
          "Header": "File Version",
          "accessor": "version",
        },
        Object {
          "Header": "Edit",
          "accessor": [Function],
          "disableSortBy": true,
        },
        Object {
          "Header": " ",
          "accessor": [Function],
          "disableSortBy": true,
        },
      ]
    }
    data={Array []}
    fetchData={[Function]}
    hasChildren={[Function]}
    identifierField="id"
    linkerField="id"
    loading={false}
    loadingComponent={[Function]}
    paginate={false}
    parentIdentifierField="parent_id"
    parentNodes={
      Array [
        undefined,
        undefined,
      ]
    }
    renderNullDataComponent={[Function]}
    resize={true}
    rootParentId={null}
    useDrillDown={false}
  >
    <div
      className="table-container mb-3"
    >
      <div
        className="table div-table"
        role="table"
        style={
          Object {
            "minWidth": "750px",
          }
        }
      >
        <div
          className="thead"
        >
          <div
            className="tr"
            key="headerGroup_0"
            role="row"
            style={
              Object {
                "display": "flex",
                "flex": "1 0 auto",
                "minWidth": "250px",
              }
            }
          >
            <div
              className="th"
              colSpan={1}
              key="thead-th-0"
              onClick={[Function]}
              role="columnheader"
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "flex": "150 0 auto",
                  "minWidth": "50px",
                  "position": "relative",
                  "width": "150px",
                }
              }
              title="Toggle SortBy"
            >
              Identifier
              <SortIcon
                isSorted={false}
              >
                <div
                  className="icon-wrapper"
                >
                  <div
                    className="previous-next-filled  icon"
                  />
                </div>
              </SortIcon>
              <div
                className="resizer "
                draggable={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="separator"
                style={
                  Object {
                    "cursor": "ew-resize",
                  }
                }
              />
            </div>
            <div
              className="th"
              colSpan={1}
              key="thead-th-1"
              onClick={[Function]}
              role="columnheader"
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "flex": "150 0 auto",
                  "minWidth": "50px",
                  "position": "relative",
                  "width": "150px",
                }
              }
              title="Toggle SortBy"
            >
              File Name
              <SortIcon
                isSorted={false}
              >
                <div
                  className="icon-wrapper"
                >
                  <div
                    className="previous-next-filled  icon"
                  />
                </div>
              </SortIcon>
              <div
                className="resizer "
                draggable={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="separator"
                style={
                  Object {
                    "cursor": "ew-resize",
                  }
                }
              />
            </div>
            <div
              className="th"
              colSpan={1}
              key="thead-th-2"
              onClick={[Function]}
              role="columnheader"
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "flex": "150 0 auto",
                  "minWidth": "50px",
                  "position": "relative",
                  "width": "150px",
                }
              }
              title="Toggle SortBy"
            >
              File Version
              <SortIcon
                isSorted={false}
              >
                <div
                  className="icon-wrapper"
                >
                  <div
                    className="previous-next-filled  icon"
                  />
                </div>
              </SortIcon>
              <div
                className="resizer "
                draggable={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="separator"
                style={
                  Object {
                    "cursor": "ew-resize",
                  }
                }
              />
            </div>
            <div
              className="th"
              colSpan={1}
              key="thead-th-3"
              role="columnheader"
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": undefined,
                  "flex": "150 0 auto",
                  "minWidth": "50px",
                  "position": "relative",
                  "width": "150px",
                }
              }
            >
              Edit
              <div
                className="resizer "
                draggable={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="separator"
                style={
                  Object {
                    "cursor": "ew-resize",
                  }
                }
              />
            </div>
            <div
              className="th"
              colSpan={1}
              key="thead-th-4"
              role="columnheader"
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": undefined,
                  "flex": "150 0 auto",
                  "minWidth": "50px",
                  "position": "relative",
                  "width": "150px",
                }
              }
            >
               
              <div
                className="resizer "
                draggable={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="separator"
                style={
                  Object {
                    "cursor": "ew-resize",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="tbody"
          role="rowgroup"
        />
      </div>
      <div
        className="jumbotron"
      >
        <p>
          No Data Found
        </p>
      </div>
    </div>
  </Table>
</DrillDownTable>
`;
